SQL/Archive/2018-05-28_publication_schema.sql

```diff
@@ -0,0 +1,160 @@
+    CONSTRAINT `FK_publication_2` FOREIGN KEY(`RatedBy`) REFERENCES `users` (`ID`),
+    CONSTRAINT `FK_publication_3` FOREIGN KEY(`PublicationStatusID`) REFERENCES `publication_status` (`PublicationStatusID`)
+) ENGINE=InnoDB DEFAULT CHARSET='utf8mb4';
+
+-- Separate table for Keywords
+DROP TABLE IF EXISTS publication_keyword;
+CREATE TABLE `publication_keyword` (
+  `PublicationKeywordID` int(10) unsigned NOT NULL AUTO_INCREMENT,
+  `Label` varchar(255) NOT NULL,
+  PRIMARY KEY(`PublicationKeywordID`)
+) ENGINE=InnoDB DEFAULT CHARSET='utf8mb4';
+
+-- Publication - Keyword relational table
+DROP TABLE IF EXISTS publication_keyword_rel;
+CREATE TABLE `publication_keyword_rel` (
+  `PublicationID` int(10) unsigned NOT NULL,
+  `PublicationKeywordID` int(10) unsigned NOT NULL,
+  CONSTRAINT `UK_publication_keyword_1` UNIQUE (PublicationID, PublicationKeywordID),
+  CONSTRAINT `FK_publication_keyword_1` FOREIGN KEY(`PublicationID`) REFERENCES `publication` (`PublicationID`),
+  CONSTRAINT `FK_publication_keyword_2` FOREIGN KEY(`PublicationKeywordID`) REFERENCES `publication_keyword` (`PublicationKeywordID`)
+) ENGINE=InnoDB DEFAULT CHARSET='utf8mb4';
+
+
+DROP TABLE IF EXISTS publication_collaborator;
+CREATE TABLE `publication_collaborator` (
+  `PublicationCollaboratorID` int(10) unsigned NOT NULL AUTO_INCREMENT,
+  `Name` varchar(255) NOT NULL,
+  `Email` varchar(255),
+  PRIMARY KEY(`PublicationCollaboratorID`),
+  CONSTRAINT `UK_publication_collaborator_Name` UNIQUE (`Name`),
+  CONSTRAINT `UK_publication_collaborator_Email` UNIQUE (`Email`)
+) ENGINE=InnoDB DEFAULT CHARSET='utf8mb4';
+
+DROP TABLE IF EXISTS publication_collaborator_rel;
+CREATE TABLE `publication_collaborator_rel` (
+  `PublicationID` int(10) unsigned NOT NULL,
+  `PublicationCollaboratorID` int(10) unsigned NOT NULL,
+  CONSTRAINT `UK_publication_collaborator_1` UNIQUE (PublicationID, PublicationCollaboratorID),
+  CONSTRAINT `FK_publication_collaborator_1` FOREIGN KEY(`PublicationID`) REFERENCES `publication` (`PublicationID`),
+  CONSTRAINT `FK_publication_collaborator_2` FOREIGN KEY(`PublicationCollaboratorID`) REFERENCES `publication_collaborator` (`PublicationCollaboratorID`)
+) ENGINE=InnoDB DEFAULT CHARSET='utf8mb4';
+
+-- Publication - Variable of Interest  relational table
+DROP TABLE IF EXISTS publication_parameter_type_rel;
+CREATE TABLE `publication_parameter_type_rel` (
+    `PublicationID` int(10) unsigned NOT NULL,
+    `ParameterTypeID` int(10) unsigned NOT NULL,
+    CONSTRAINT `UK_publication_parameter_type_1` UNIQUE (PublicationID, ParameterTypeID),
+    CONSTRAINT `FK_publication_parameter_type_rel_1` FOREIGN KEY (`PublicationID`) REFERENCES `publication` (`PublicationID`),
+    CONSTRAINT `FK_publication_parameter_type_rel_2` FOREIGN KEY (`ParameterTypeID`) REFERENCES `parameter_type` (`ParameterTypeID`)
+) ENGINE=InnoDB DEFAULT CHARSET='utf8mb4';
+
+DROP TABLE IF EXISTS publication_test_names_rel;
+CREATE TABLE `publication_test_names_rel` (
```

Comment:
I would add a Primary key made of the 2 columns. Unless duplicates are expected.

Reply:
done.